src/taskExecutor.ts

```diff
@@ -0,0 +1,40 @@
+class Queue<T> {
+    private _store: T[] = [];
+    public push(val: T): void {
+        this._store.push(val);
+    }
+    public pop(): T | undefined {
+        return this._store.shift();
+    }
+}
+
+class TaskExecutor {
+    private _tasks: Queue<any> = new Queue();

```

Comment:
Add `dispose()` function which clear the queue when deactivating

Reply:
added.